packages/gatsby/src/redux/index.ts

```diff
@@ -37,6 +37,12 @@ export const readState = (): IGatsbyState => {
       })
     }
 
+    if (!state.telemetry) {
+      state.telemetry = {
+        gatsbyImageSourceUrls: new Set<string>(),
+      }
+    }
```

Comment:
This shouldn't be necessary. Did you have an example where this fails without it?

Reply:
My thought was that I didn't want to overwrite anything if it already existed, but since we don't persist `telemetry` to disk, than this would always be empty. Is that a correct assumption?

Reply:
Your reducer shoulds et the initial state, nothing else

Reply:
Got it :+1: